xrpl/utils/txn_parser/utils/order_book_parser.py

```diff
@@ -0,0 +1,217 @@
+    currency_amount: Union[str, Dict[str, str]]
+) -> CurrencyAmount:
+    if isinstance(currency_amount, str):
+        return CurrencyAmount(currency="XRP", value=str(drops_to_xrp(currency_amount)))
+    else:
+        return CurrencyAmount(
+            currency=currency_amount["currency"],
+            issuer=currency_amount["issuer"],
+            value=currency_amount["value"],
+        )
+
+
+def _calculate_delta(
+    final_amount: CurrencyAmount,
+    previous_amount: CurrencyAmount,
+) -> str:
+    final_value = get_value(final_amount)
+    previous_value = get_value(previous_amount)
+    delta = final_value - previous_value
+    return str(delta)
+
+
+def _get_change_amount(
+    node: NormalizedNode,
+    status: Literal["created", "partially-filled", "filled", "cancelled"],
+    side: Literal["TakerGets", "TakerPays"],
+) -> Optional[CurrencyAmount]:
+    if status == "cancelled":
+        final_fields = node.get("FinalFields")
+        if final_fields is not None:
+            final_fields_amount = final_fields.get(side)
+            if final_fields_amount is not None:
+                return _derive_currency_amount(final_fields_amount)
+    if status == "created":
+        new_fields = node.get("NewFields")
+        if new_fields is not None:
+            new_fields_amount = new_fields.get(side)
+            if new_fields_amount is not None:
+                return _derive_currency_amount(new_fields_amount)
+    final_fields = node.get("FinalFields")
+    previous_fields = node.get("PreviousFields")
+    if final_fields is not None and previous_fields is not None:
+        final_fields_amount = final_fields.get(side)
+        previous_fields_amount = previous_fields.get(side)
+        if final_fields_amount is not None and previous_fields_amount is not None:
+            final_amount = _derive_currency_amount(final_fields_amount)
+            previous_amount = _derive_currency_amount(previous_fields_amount)
+            value = _calculate_delta(final_amount, previous_amount)
+            # final_amount is being reused because it makes
+            # it easier to return the changed amount.
+            final_amount["value"] = value
+            # From now on you could consider final_amount as changed_amount.
+            return final_amount
+    return None
+
+
+def _get_quality(
+    taker_gets: CurrencyAmount,
+    taker_pays: CurrencyAmount,
+    book_directory: str,
+) -> str:
+    quality_hex = book_directory[-16:]  # last 16 characters of the BookDirectory
+    taker_gets_currency = taker_gets["currency"]
+    taker_pays_currency = taker_pays["currency"]
+    mantissa = float.fromhex(quality_hex[2:])
+    offset = int(quality_hex[:2], 16) - 100
+    scientific_quality = "{:.16f}e{}".format(mantissa, offset)
+    quality = Decimal(scientific_quality)
+    if taker_gets_currency == "XRP" or taker_pays_currency == "XRP":
+        quality = quality * 1000000
+    return str(quality.normalize())
+
+
+def _get_optional_fields(
```

Comment:
It's not obvious to me from the name of this function why `ModifiedFields` would not be checked - would it be possible to add an explaining comment diving into more detail for what this function should be used for? (Or maybe update the name?)